docs/website/contents/for-developers/NodeTasks.md

```diff
@@ -0,0 +1,303 @@
+# Overview of the tasks of a caught-up node
+
+This document gives an overview of the tasks of a caught-up node, both as a relay and as a block producer.
+
+## In a single node
+
+We refer to the [component diagram](./ComponentDiagram.md) for a very high-level overview of the data flow in a node.
+Another good diagram is [Figure 1.1. in the Shelley networking spec](https://ouroboros-network.cardano.intersectmbo.org/pdfs/network-spec/network-spec.pdf).
+
+Also see the [Glossary](./Glossary.md) for jargon explanations.
+
+### Chain selection and storage layer
+
+As a basis for all other operations, the node needs to maintain a *current selection*, ie the best valid chain it has seen so far.
+
+The selection is made up of an immutable and a volatile part:
+
+ - The immutable part of the selection consists of blocks that are no longer subject to rollback.
+   This is the case when a block has been confirmed by at least $k$ (the *security parameter*, $k=2160$ on mainnet) blocks.
+   Immutable blocks stored in the ImmutableDB.
+
+ - The volatile part of the selection are the newest $k$ blocks.
+   They are stored in the VolatileDB, together with other blocks that could be on a fork we might switch to in the future.
+
+Additionally, the LedgerDB contains the ledger state corresponding to all points on the volatile part of the chain as well as the tip of the immutable chain, in order to validate new blocks and potential forks.
+
+The flow of information is depicted in the following diagram.
+Rectangular boxes stand for logical components, and hexagons correspond to Haskell RTS threads.
+
+```mermaid
+graph LR
+  subgraph ChainSelectionAndStorageLayer[ChainSelection and storage layer]
+    subgraph ChainSelection[ChainSelection]
+      direction TB
+
+      CurrentSelection["`**CurrentSelection**`"]
+      AddBlockRunner{{AddBlockRunner}}
+
+      AddBlockRunner -- maintains --> CurrentSelection
+    end
+
+    ChainSelection -- "backed by" --> StorageLayer
+
+    subgraph StorageLayer[StorageLayer]
+      direction TB
+
+      VolatileDB["`**VolatileDB**`"]
+      ImmutableDB["`**ImmutableDB**`"]
+      LedgerDB["`**LedgerDB**`"]
+
+      CopyToImmutableDB{{CopyToImmutableDB}}
+      GarbageCollect{{GarbageCollect}}
+      CreateLedgerSnapshot{{CreateLedgerSnapshot}}
+
+      VolatileDB --> CopyToImmutableDB
+      CopyToImmutableDB --> ImmutableDB
+      GarbageCollect -- "cleans up" --> VolatileDB
+      LedgerDB --> CreateLedgerSnapshot
```

Comment:
This looks like it is aligned with the ImmutableDB, but it should be aligned with the volatiledb. Maybe it is not possible with mermaid?

Reply:
Yeah, I don't think Mermaid has enough flexibility for stuff like this. Maybe it is possible to do sth with invisible arrows, but I don't think it is *too* terrible.